backend/core/src/secondary-adapters/postgres/history/history-repository-adapter.ts

```diff
@@ -0,0 +1,29 @@
+import { inject, injectable } from 'inversify';
+import { HistoryRepository } from '~/core/history/port/history-repository';
+import { PrismaClient, History } from '@prisma/client';
+import { CONTAINER_TYPES, HistoryResponseDto, HistoryRequestDto } from '~/shared/types/types';
+
+@injectable()
+export class HistoryRepositoryAdapter implements HistoryRepository {
+  private prismaClient: PrismaClient;
+
+  constructor(@inject(CONTAINER_TYPES.PrismaClient) prismaClient: PrismaClient) {
+    this.prismaClient = prismaClient;
+  }
+
+  getAllUserHistory(userId: string): Promise<History[]> {
+    return this.prismaClient.history.findMany({
+      where: {
+        userId,
+      },
+    });
+  }
+
+  async createHistory(historyRequestDto: HistoryRequestDto): Promise<HistoryResponseDto> {
```

Comment:
maybe rename the method to createHistoryItem or something like this to make it more clear? (the same goes for controller and service)

Reply:
Done!